src/main/java/name/nkid00/teaminess/mixin/ChatMessageMixin.java

```diff
@@ -0,0 +1,71 @@
+package name.nkid00.teaminess.mixin;
+
+import name.nkid00.teaminess.Teaminess;
+import name.nkid00.teaminess.model.Waypoint;
+import name.nkid00.teaminess.model.Location;
+import net.minecraft.network.packet.c2s.play.ChatMessageC2SPacket;
+import net.minecraft.server.network.ServerPlayNetworkHandler;
+import net.minecraft.server.network.ServerPlayerEntity;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.InvalidIdentifierException;
+import net.minecraft.util.Pair;
+import net.minecraft.util.math.BlockPos;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import static java.lang.Integer.parseInt;
+
+@Mixin(ServerPlayNetworkHandler.class)
+public class ChatMessageMixin {
```

Comment:
Why don't we use [fabric api](https://maven.fabricmc.net/docs/fabric-api-0.58.5+1.19.1/net/fabricmc/fabric/api/message/v1/ServerMessageEvents.html#CHAT_MESSAGE) but invent a wheel?